RFC/src/RFC-0140 Syncing and seeding.md

```diff
@@ -0,0 +1,131 @@
+   disclaimer in the documentation and/or other materials provided with the distribution.
+3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products
+   derived from this software without specific prior written permission.
+
+THIS DOCUMENT IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", 
+"NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in 
+[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all capitals, as 
+shown here.
+
+## Disclaimer
+
+The purpose of this document and its content is for information purposes only and may be subject to change or update
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community regarding the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This document describes the process of Syncing, seeding, pruning and cut-through.
+
+## Related RFCs
+
+* [RFC-0110: Base Nodes](RFC-0110_BaseNodes.md)
+
+## Descriptions
+
+### Syncing
+
+When a new node comes online, looses connection or encounters a fork that is longer than is longer than it can solve, it must enter syncing mode. This will allow it to recover its state to the newest up to date state. Sycning can be diveded into 2 [SynchronisationStrategy]s, complete sync and sync. Complete sync will involve that the node communicates with an archive node to get the complete history of every single block from genesis block. Sync will involve the node getting every block from its [puning horizon](pruninghorizon) to current head, as well as every block header from genesis block. 
+
+#### Complete Sync
+
+Complete sync is only available from archive nodes, as these will be the only nodes that will be able to supply the complet history required to sync every block with every transaction from genesis block up onto [current head](currenthead). 
+
+
+
+#### Syncing process
```

Comment:
In Grin the fast sync involves 
- Download all block headers on the most worked chain
- Download the full state as it was at the node/pruning horizon, including the full UTXO, range proof and kernel data, as well as all corresponding MMRs. 
- Validate the full state [_**Note:** The entire pruned blockchain up to the node/pruning horizon is validated_]
- Download full blocks since the horizon to get to the chain head

I do not see this in our scheme and believe we should have something similar, or am I missing something?

Reply:
> * Validate the full state [_**Note:** The entire pruned blockchain up to the node/pruning horizon is validated_]

are saying we should validate each and every snapshot of the blockchain since genesis?

Currently we validate the snapshot of the UTXO has horizon, and then validate each tx as we move forward

Reply:
I think we should validate the full state of the blockchain (which equals the pruned blockchain) up to the node horizon. Same as Grin. And then all blocks from there to the tip.

Reply:
see top comment?